src/ctap/mod.rs

```diff
@@ -657,11 +682,14 @@ where
         let credentials = if let Some(allow_list) = allow_list {
             let mut found_credentials = vec![];
             for allowed_credential in allow_list {
-                match self
-                    .persistent_store
-                    .find_credential(&rp_id, &allowed_credential.key_id)
-                {
-                    Some(credential) => found_credentials.push(credential),
+                match self.persistent_store.find_credential(
+                    &rp_id,
+                    &allowed_credential.key_id,
+                    !has_uv,
```

Comment:
I guess CTAP doesn't require the authenticator to return a different error whether the credential does not exist and whether it is protected. But maybe this question doesn't make sense, because it looks like asking for user verification does not depend on which credential we need (it's always done upfront).

Reply:
External user Verification is done with another command before the actual request. Internal user verification like a fingerprint reader on the security key happens while processing a command, like waiting for user presence.

In this case, it shouldn't really matter since the point of credProtect is to make it indistinguishable whether the credential is protected or does not exist.